### PR TITLE
Fix build_solr_schema to match production. [PD-1703]

### PR DESCRIPTION
**Pull request by darrint**
This isolates some changes from the previous api-merge branch to make build_solr_schema output the exact same schema as what is in deployment/search/.

So far the only testing is to make sure the output matches.

**Commentary Requested**: Can anyone think of a reason not to merge this? Will there be any odd side effects from these changes?

### CODE REVIEW

File: seo/search_backend.py [hunk 1]
```diff
@@ -482,7 +482,10 @@ def build_schema(self, fields):
             field_mapper = {'datetime': 'date',
                             'integer': 'long',
                             'string': 'text_en'}
-            if field_class.field_type:
+            if getattr(field_class, 'trust_field_type', False):
```
Comment (aspidites) [inline seo/search_backend.py]: Any reason to do a safe getattr for `trust_field_type`, but not for `field_type`? Is `field_type` guaranteed to exist?

File: seo/search_indexes.py [hunk 2]
```diff
@@ -42,21 +42,16 @@ def __init__(self, *args, **kwargs):
...
-    not everything matches up with the model. Doesn't work for indexed fields,
-    likely because of some setting in search_backends.py.
-
-    """
-    field_type = 'string'
-
-
-class MultiValueIntegerField(indexes.MultiValueField):
-        field_type = 'int'
+class MultiValueSpecialField(indexes.MultiValueField):
```
Comment (aspidites) [inline seo/search_indexes.py]: Missing documentation.

File: seo/search_backend.py (hunk 1 above)
Comment (aspidites) [inline seo/search_backend.py]: Looking below, it seems the field only exists for `MultiValueSpecialField`s.

File: seo/search_indexes.py [hunk 3]
```diff
@@ -42,21 +42,16 @@ def __init__(self, *args, **kwargs):
...
+    def __init__(self, *args, **kwargs):
+        self.field_type = kwargs.pop('field_type', 'text_en')
```
Comment (aspidites) [inline seo/search_indexes.py]: Are these custom attributes meant to sandwich the call to super?

**Comment by aspidites** [comment]
From my comments [here]():

```
I had to remove a couple of unusable options:

"side = 'true'" :
sed -i -e '/side=\"front\"/d' ~/solr-5.1.0/server/solr/{api,api_test,myjobs,myjobs_test,myjobs_test_current,seo,seo_test}/conf/schema.xml
"enablePositionIncrements = 'true'":
sed -i -e '/enablePositionIncrements=\"true\"/d' ~/solr-5.1.0/server/solr/{api,api_test,myjobs,myjobs_test,myjobs_test_current,seo,seo_test}/conf/schema.xml
```

In particular, it looks like the `enablePositionIncrements` thing is taken care of, but what about `side = 'true'`?

File: templates/search_configuration/solr.xml [hunk 4]
```diff
@@ -189,7 +187,7 @@
         <tokenizer class="solr.WhitespaceTokenizerFactory" />
         <filter class="solr.LowerCaseFilterFactory" />
         <filter class="solr.WordDelimiterFilterFactory" generateWordParts="1" generateNumberParts="1" catenateWords="0" catenateNumbers="0" catenateAll="0" splitOnCaseChange="1"/>
-        <filter class="solr.EdgeNGramFilterFactory" minGramSize="2" maxGramSize="15" side="front" />
+        <filter class="solr.EdgeNGramFilterFactory" minGramSize="2" maxGramSize="15" />
       </analyzer>
       <analyzer type="query">
```
Comment (darrint) [inline templates/search_configuration/solr.xml]: This change removes `side=true`.

**Comment by aspidites** [comment]
I'll pull this locally eventually to see if I can replicate a functional schema file. Confident with what you explained in real life though. Good work.

**Comment by JLMcLaugh** [comment]
Something isn't right. The "description" field changed from text_and_symbols to text_en type.

**Comment by darrint** [comment]
I don't see a change. What are you comparing against?

**Comment by darrint** [comment]
This is working for me and @aspidites. We're attributing the problem @JLMcLaugh saw to some kind of unknown funk in his development environment.

**Comment by JLMcLaugh** [comment]
Turns out the field issues were on my end. An old binary in myprofiles was adding extra fields.

**Pull request merged**